app/routes/public/speakers.js

```diff
@@ -30,25 +30,69 @@ export const SPEAKERS_FILTER = [
 
 @classic
 export default class SpeakersRoute extends Route {
-  async model() {
+  queryParams = {
+    search: {
+      refreshModel: true
+    }
+  }
+
+  titleToken() {
+    return this.l10n.t('Speakers');
+  }
+
+  async model(params) {
     const eventDetails = this.modelFor('public');
+    const filterOptions = [
+      {
+        and: [
+          {
+            name : 'sessions',
+            op   : 'any',
+            val  : {
+              and: [
+                {
+                  name : 'deleted-at',
+                  op   : 'eq',
+                  val  : null
+                },
+                {
+                  or: [{
+                    name : 'state',
+                    op   : 'eq',
+                    val  : 'accepted'
+                  }, {
+                    name : 'state',
+                    op   : 'eq',
+                    val  : 'confirmed'
+                  }]
+                }
+              ]
+            }
+          }
+        ]
+      }
+    ];
+
+    if (params.search) {
+      filterOptions.push({
+        name : 'name',
+        op   : 'ilike',
+        val  : `%${params.search}%`
+      });
+    }
     return {
       event    : eventDetails,
       speakers : await this.infinity.model('speakers', {
-        filter        : SPEAKERS_FILTER,
-        perPage       : 9,
-        startingPage  : 1,
-        perPageParam  : 'page[size]',
-        pageParam     : 'page[number]',
-        store         : eventDetails,
-        include       : 'sessions.track',
-        sort          : 'order',
-        cache         : true,
-        public        : true,
-        infinityCache : 36000
+        filter       : filterOptions,
+        perPage      : 9,
+        startingPage : 1,
+        perPageParam : 'page[size]',
+        pageParam    : 'page[number]',
+        store        : eventDetails,
+        include      : 'sessions.track',
+        sort         : 'order',
+        public       : true
```

Comment:
Why removed cache?

Reply:
the filter only works on reload due to caching